dune/grid/cpgrid/Entity.hpp

```diff
@@ -331,14 +348,14 @@ namespace Dune
 //          }
 
             /// Const member by pointer operator.
-            Entity* operator->() const
+            const Entity* operator->() const
             {
                 assert(Entity::valid());
                 return const_cast<EntityPointer*>(this); // const_cast-hack added because of error in vtkwriter.hh
```

Comment:
If this hack is not needed anymore (I don't remember the vtkwriter bug but it has probably been fixed) we should remove the const_cast, at the same time you change the method signature. Same for the `operator*()`.